src/ts/shared/schemaType/EmbeddedType.ts

```diff
@@ -0,0 +1,63 @@
+import { cloneDeep, isPlainObject, set as setIn } from 'lodash';
+import type { MvRecord } from '#shared/types';
+import { ensureArray } from '#shared/utils';
+import Document from '../../Document';
+import type Schema from '../../Schema';
+import BaseSchemaType from './BaseSchemaType';
+
+/** Embedded Schema Type */
+class EmbeddedType extends BaseSchemaType {
+	/** An instance of Schema representing the the document structure of embedded object contents */
+	private valueSchema: Schema;
+
+	public constructor(valueSchema: Schema) {
+		super();
+
+		this.valueSchema = valueSchema;
+	}
+
+	/**
+	 * Cast to embedded data type
+	 * @throws {@link TypeError} Throws if a non-null/non-object is passed
+	 */
+	public override cast(value: unknown): Document {
+		// convert value to a plain structure and then recast as embedded document
+		const plainValue = value == null ? {} : JSON.parse(JSON.stringify(value));
+		if (!isPlainObject(plainValue)) {
+			throw new TypeError('Cast value must be an object');
+		}
+		return new Document(this.valueSchema, { data: plainValue, isSubdocument: true });
+	}
+
+	/** Get value from mv data */
+	public get(record: MvRecord): Document {
+		const embeddedDocument = new Document(this.valueSchema, { isSubdocument: true, record });
+		return embeddedDocument;
+	}
+
+	/** Set specified embedded document value into mv record */
+	public set(originalRecord: MvRecord, setValue: Document): MvRecord {
+		const record = cloneDeep(originalRecord);
+		const subrecord = setValue.transformDocumentToRecord();
+		subrecord.forEach((value, arrayPos) => {
+			if (typeof value !== 'undefined') {
+				setIn(record, [arrayPos], value);
+			}
+		});
+		return record;
+	}
+
+	/** Validate the embedded document */
+	public async validate(document: Document): Promise<string[]> {
```

Comment:
Similarly here this was returning `Object[]` which was different than the base class, so it was adjusted.